### PR TITLE
fix(efs): fix bug when setting both lifecyclePolicy and outOfInfrequentAccessPolicy

### LINKED ISSUE #19058: (aws-efs): EFS FileSystem construction always fails when passing both lifecyclePolicy and outOfInfrequentAccessPolicy
### What is the problem?

When deploying [an example in the official CDK document](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_efs-readme.html), EFS creation always fails with the following error:

```text
One or more LifecyclePolicy objects specified are malformed.
```

After some investigation, I have found that setting both `lifecyclePolicy` and `outOfInfrequentAccessPolicy` is the cause of the problem. When you pass only one of them to a construct, it successfully deployed to AWS.

I assume that means CloudFormation template does not allow multiple entries in one `LifecyclePolicies` item.  So, I have not confirmed though, the following changes to emitted CloudFormation template may work.

### Bad

```json
"LifecyclePolicies": [
  {
    "TransitionToIA": "AFTER_14_DAYS",
    "TransitionToPrimaryStorageClass": "AFTER_1_ACCESS"
  }
],
```

### Good

```json
"LifecyclePolicies": [
  {
    "TransitionToIA": "AFTER_14_DAYS"
  },
  {
    "TransitionToPrimaryStorageClass": "AFTER_1_ACCESS"
  }
],
```

### Running the example code in the official document

CDK stack source code: 

```ts
const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
  vpc: new ec2.Vpc(this, 'VPC'),
  lifecyclePolicy: efs.LifecyclePolicy.AFTER_14_DAYS, // files are not transitioned to infrequent access (IA) storage by default
  performanceMode: efs.PerformanceMode.GENERAL_PURPOSE, // default
  outOfInfrequentAccessPolicy: efs.OutOfInfrequentAccessPolicy.AFTER_1_ACCESS, // files are not transitioned back from (infrequent access) IA to primary storage by default
});
```

CloudFormation template emitted:

```jsonc
// -- snip --
    "MyEfsFileSystem570A3D32": {
      "Type": "AWS::EFS::FileSystem",
      "Properties": {
        "Encrypted": true,
        "FileSystemTags": [
          {
            "Key": "Name",
            "Value": "CdkTestStack/MyEfsFileSystem"
          }
        ],
        "LifecyclePolicies": [
          {
            "TransitionToIA": "AFTER_14_DAYS",
            "TransitionToPrimaryStorageClass": "AFTER_1_ACCESS"
          }
        ],
        "PerformanceMode": "generalPurpose"
      },
      "UpdateReplacePolicy": "Delete",
      "DeletionPolicy": "Delete",
      "Metadata": {
        "aws:cdk:path": "CdkTestStack/MyEfsFileSystem/Resource"
      }
    },
// -- snip --
```

### What did you expect to happen?

You can successfully deploy EFS even when you set both `lifecyclePolicy` and `outOfInfrequentAccessPolicy`.

### What actually happened?

Deployment fails with the error message:

```text
One or more LifecyclePolicy objects specified are malformed.
```

### CDK CLI Version

2.13.0 (build b0b744d)

### Framework Version

_No response_

### Node.js Version

v16.13.2

### OS

macOS 12.2.1

### Language

Typescript

### Language Version

4.5.5

### Other information

_No response_

### PR DESCRIPTION
**Pull request by robertd**
Closes #19058

Most likely this was an oversight in CloudFormation docs. Submitted a PR for CF docs here: https://github.com/awsdocs/aws-cloudformation-user-guide/pull/1165

Here are the API docs showing the proper parameters for LifecyclePolicies: https://docs.aws.amazon.com/efs/latest/ug/API_PutLifecycleConfiguration.html

image
Screen Shot 2022-02-21 at 5 14 50 PM
Screen Shot 2022-02-21 at 5 14 57 PM

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by robertd** [comment]
Hi @corymhall.  This PR is ready for your review. TIA.

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 1]
```diff
@@ -341,15 +341,29 @@ export class FileSystem extends FileSystemBase {
     const encrypted = props.encrypted ?? (FeatureFlags.of(this).isEnabled(
       cxapi.EFS_DEFAULT_ENCRYPTION_AT_REST) ? true : undefined);
 
+    // LifecyclePolicies is an array of lists containing a single policy
+    let lifecyclePolicies = [];
+
+    const lifecyclePolicy = props.lifecyclePolicy ? {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:347]: Can this be simplified to just

```ts
if (props.lifecyclePolicy) {
  lifecyclePolicies.push({ transitionToIa: props.lifecyclePolicy });
}
```

Same for `outOfInfrequentAccessPolicy`

**Review by corymhall — DISMISSED** [review]
@robertd thanks for this fix! Just one minor comment.

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts (hunk 1 above)
Comment (robertd) [reply packages/@aws-cdk/aws-efs/lib/efs-file-system.ts]: Done.

**Review by corymhall — APPROVED** [review]

**Pull request merged**